scss/forms/_input-group.scss

```diff
@@ -34,11 +34,21 @@ $input-prefix-padding: 1rem !default;
     }
 
     > :first-child {
-      border-radius: if($global-text-direction == rtl, 0 $input-radius $input-radius 0, $input-radius 0 0 $input-radius);
+      &:not(.input-group-button){
```

Comment:
Foundation provide a SASS library, not only CSS components.
* We can't expect Input group button to be implemented as `.input-group-button` (user should be able to create custom components with input group mixins)
* Why only targetting Input group button ?

So I would rather go for:
```scss
&, &  > * { ... }
```

To be honnest, the `> *` is ugly and should be removed. But we'll keep it for backward compatibility. Adding a comment here like "don't do this" could be useful so it does not became an example.

Reply:
Since `.input-group-button` is hardcoded later in the `foundation-form-prepostfix` mixin, can't we expect it to be implemented as `.input-group-button`?

Reply:
Ah f**. I did not realized that Input Group was not _component-oriented_ at all !
Well, its still a problem: Why only targetting Input group button ?

This is generally a bad idea to make our CSS component having particular effects on each-others as the user should be able to use them with custom CSS alongside its own components. Input Group should have the same behavior no matter what is the class of the button we put it. This is the best way to make someone loose 1 hour: "Why when I put my own button there it is not rounded anymore."